ui-lib/src/main/java/co/electriccoin/zcash/ui/common/datasource/NearSwapDataSourceImpl.kt

```diff
@@ -0,0 +1,146 @@
+        slippage: BigDecimal,
+    ): SwapQuote {
+        val decimals =
+            when (swapMode) {
+                SwapMode.EXACT_INPUT -> originAsset.decimals
+                SwapMode.EXACT_OUTPUT -> destinationAsset.decimals
+            }
+
+        val shifted = amount.movePointRight(decimals)
+        val integer = shifted.toBigInteger().toBigDecimal()
+        val normalizedAmount = shifted.round(MathContext(integer.precision(), RoundingMode.HALF_EVEN))
+
+        val request =
+            QuoteRequest(
+                dry = false,
+                swapType =
+                    when (swapMode) {
+                        SwapMode.EXACT_INPUT -> SwapType.EXACT_INPUT
+                        SwapMode.EXACT_OUTPUT -> SwapType.EXACT_OUTPUT
+                    },
+                slippageTolerance = slippage.multiply(BigDecimal(100), MathContext.DECIMAL128).toInt(),
+                originAsset = originAsset.assetId,
+                depositType = RefundType.ORIGIN_CHAIN,
+                destinationAsset = destinationAsset.assetId,
+                amount = normalizedAmount,
+                refundTo = originAddress,
+                refundType = RefundType.ORIGIN_CHAIN,
+                recipient = destinationAddress,
+                recipientType = RecipientType.DESTINATION_CHAIN,
+                deadline = Clock.System.now() + 10.minutes,
+                quoteWaitingTimeMs = QUOTE_WAITING_TIME,
+                appFees =
+                    listOf(
+                        AppFee(
+                            recipient = "electriccoinco.near",
+                            fee = AFFILIATE_FEE_BPS
+                        )
+                    )
+            )
+
+        return try {
+            NearSwapQuote(nearApiProvider.requestQuote(request))
+        } catch (e: ResponseWithErrorException) {
+            when {
+                e.error.message.startsWith("Amount is too low for bridge, try at least") -> {
+                    val errorAmount =
+                        e.error.message
+                            .split(" ")
+                            .lastOrNull()
+                            ?.toBigDecimalOrNull() ?: throw e
+                    val errorAsset =
+                        when (swapMode) {
+                            SwapMode.EXACT_INPUT -> originAsset
+                            SwapMode.EXACT_OUTPUT -> destinationAsset
+                        }
+                    throw QuoteLowAmountException(
+                        asset = errorAsset,
+                        amount = errorAmount,
+                        amountFormatted = errorAmount.movePointLeft(errorAsset.decimals)
+                    )
+                }
+
+                e.error.message.startsWith("No quotes found") -> throw QuoteLowAmountException(
+                    asset = originAsset,
+                    amount = null,
+                    amountFormatted = null
+                )
+
+                else -> throw e
```

Comment:
You bad boy! Our linter would kill me, burn xcode and make macbook dance. At least 3 chars are must throughout the code :) (err typically)